From 69a73ba3d4f1a93cc1824b2d79004746818dcb9d Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 4 Feb 2015 16:17:22 -0800
Subject: [PATCH] force setting of MODULE_SYNCQT_DIR correctly,

Fixes errors like

Project ERROR: Failed to run: perl -w
/home/ubuntu/work/daisy/build-xi3v2/tmp/sysroots/x86_64-linux/usr/bin/qt5/syncqt.pl
-module QtWebKit -version 5.1.1 -outdir
/home/ubuntu/work/daisy/build-xi3v2/tmp/work/xi3v2-rdk-linux/qtwebkit/5.1.1-r0/build
/home/ubuntu/work/daisy/build-xi3v2/tmp/work/xi3v2-rdk-linux/qtwebkit/5.1.1-r0/git/source/qtwebkit
ERROR: Error calling
/home/ubuntu/work/daisy/build-xi3v2/tmp/sysroots/x86_64-linux/usr/bin/qt5/qmake
-makefile -o Makefile  -r  QMAKE_CFLAGS+=-g0 QMAKE_CXXFLAGS+=-g0
DEFINES+=BCM_QTWEBKIT=1
/home/ubuntu/work/daisy/build-xi3v2/tmp/work/pacexi3v2-rdk-linux/qtwebkit/5.1.1-r0/git/source/qtwebkit/WebKit.pro

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 source/qtbase/mkspecs/features/qt_module_headers.prf | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

Index: qtbase/mkspecs/features/qt_module_headers.prf
===================================================================
--- qtbase.orig/mkspecs/features/qt_module_headers.prf
+++ qtbase/mkspecs/features/qt_module_headers.prf
@@ -10,6 +10,21 @@
 #
 
 load(qt_build_paths)
+# FIXME
+# STM_SGP_HH : tmp workaround to force setting of MODULE_SYNCQT_DIR correctly,
+#              which should be done in qt_build_paths. But for some reason,
+#              it is not loaded, probably due to existing load.
+#              as result, MODULE_SYNCQT_DIR will take value of $$MODULE_BASE_INDIR
+#              from previous load, as MODULE_BASE_DIR is set only later in api.pri
+#              Not sure how Xi4 master is doing it for now, to look at when time
+#              permits.
+isEmpty(_QMAKE_CONF_): error("Project has no top-level .qmake.conf file.")
+MODULE_BASE_INDIR = $$dirname(_QMAKE_CONF_)
+MODULE_BASE_OUTDIR = $$shadowed($$MODULE_BASE_INDIR)
+!isEmpty(MODULE_BASE_DIR): MODULE_SYNCQT_DIR = $$MODULE_BASE_DIR    # compat for webkit
+isEmpty(MODULE_SYNCQT_DIR): MODULE_SYNCQT_DIR = $$MODULE_BASE_INDIR
+isEmpty(MODULE_QMAKE_OUTDIR): MODULE_QMAKE_OUTDIR = $$MODULE_BASE_OUTDIR
+
 
 !build_pass:git_build {
     qtPrepareTool(QMAKE_SYNCQT, syncqt)
